### PR TITLE
ICD-10 Reference

### PR DESCRIPTION
**Pull request by mkorvink**
https://duck.co/ia/view/international_classification_for_diseases_icd

### CODE REVIEW

**Comment by moollaza** [comment]
@mkorvink Thanks for contributing and welcome to DuckDuckHack!

Did you happen to create an IA Page or this? If not, it's no problem, please create one and paste the link into your PR Description above.

> More details: http://docs.duckduckhack.com/submitting/pull-request.html

You just need to go to https://duck.co/ia/new_ia -- and you should be able to use the small "Skip this step" button :+1:

**Comment by moollaza** [comment]
Once that's done, my colleague @jdorweiler will be able to test your code and give feedback if needed.

Cheers!

**Comment by mkorvink** [comment]
Thanks @moollaza! Apologies for the trouble. I created an Instant Answer page and added it to the description. Please let me know if there's anything else that I've missed. I'm a first-timer so thanks for showing me the ropes!

**Comment by jdorweiler** [comment]
@mkorvink I installed it on beta for testing.  You should be able to trigger it using `icd` or `icd10`.  Are there any other trigger words you think would be useful? 
https://beta.duckduckgo.com/?q=icd+a001&ia=about

I renamed the IA page to make the id a little shorter. 
https://duck.co/ia/view/international_classification_for_diseases_icd

Can you change the directory name from `share/fathead/icd_10_cm_codes/` to `share/fatheadinternational_classification_for_deseases_icd/`

**Comment by mkorvink** [comment]
Thanks @jdorweiler. Updated the directory and submitted a new pull request. I think that the trigger words that you suggested will be just fine.  BTW: Is there a way to strip decimals out of the remainder?  Depending on the source, some ICD codes will contain decimals and others won't.  If we could strip them out, we're guaranteed a hit.  If there's a way to do it on my end just let me know and I'll give it a go!  

Thanks much for the help!

**Comment by jdorweiler** [comment]
@mkorvink not sure what you mean about the remainder.  Can you give me an example?

**Comment by mkorvink** [comment]
@jdorweiler basically ICD code itself (after the trigger word has been removed).  For example if someone searches "ICD M54.2", can we return the results for M542 (without the decimal)?

**Comment by jdorweiler** [comment]
@mkorvink I think you're looking for a redirect entry. 
http://docs.duckduckhack.com/resources/fathead-overview.html

The search term would be "ICD M54.2" and it would have a redirect to "ICD M54".  You would have to figure out which decimals make sense to add redirects for though.  Let me know if that makes sense.

**Comment by jdorweiler** [comment]
@mkorvink do the redirects make sense or do you need any help?

**Comment by mkorvink** [comment]
Thanks @jdorweiler and my apologies for the hold-up.  I'm digging into the changes now and will certainly let you know if I run into any trouble!

**Comment by mkorvink** [comment]
@jdorweiler - Added support for both procedure codes (as well diagnosis codes) and also added redirects for the diagnosis codes (to handle decimals). Please let me know if there's anything else that I'm missing.  

Thanks!

**Comment by jdorweiler** [comment]
@mkorvink I'm getting an error when I try to run the parse script.  I check the download directory and the file is missing. 
```
Traceback (most recent call last):
  File "parse.py", line 11, in <module>
    pcs_input_file = open(pcs_inp, "r")
IOError: [Errno 2] No such file or directory: 'download/icd10pcs_codes_2016.txt'
```

**Comment by mkorvink** [comment]
Thanks @jdorweiler. I pushed an update to correct the file location.  Apologies for the trouble!

**Comment by mkorvink** [comment]
BTW: It ran locally for me and produced an output file.

**Comment by jdorweiler** [comment]
@mkorvink thanks it run fine for me now.  Your format for the redirects is pretty close but we need to change it a little.  It should look like this
```
word you want to trigger on \t R \t entry that you want to show
```
A specific example.  Lets say you want the query `A001` to show the entry for `A00.1` then that entry would be:
```
A001 \t R \t A00.1
``

You would need one line for each redirect you want to have.

**Comment by mkorvink** [comment]
@jdorweiler - Apologies for the delay and thanks for the help.  I updated the script to handle the decimals.  Now if someone searches for a diagnosis code, with or without decimals, the code with decimals will be shown.  Procedure codes (the second part of the script) do not have decimals in them so no redirect is needed.  Let me know if there's anything else!

**Comment by jdorweiler** [comment]
@mkorvink are these lines in the output file where you want to have a redirect?
selection_232

So in the above line the search `A009` should show the entry for 'A00.9`?

If that's the case then you can remove the description and put them on the same line in the file.  Then use type `r` so we know it is a redirect.
`A009 r A00.9`

**Comment by moollaza** [comment]
@mkorvink thanks again for creating the ICD-10 Fathead! You’ve been doing a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in Diseases and Medical Classifications.

When you're ready to move on, please: 

- Create a new Instant Answer for searches you do related to Diseases and Medical Classifications

OR

- Test live Disease/Medical Classification Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

**Comment by mkorvink** [comment]
Hi @moollaza - Thanks for checking in! Things have been busy during my day job so I'm just getting back into the swing of things.  I should have new commits coming your way soon!

Thanks much!
Mike

**Comment by mkorvink** [comment]
Hey @jdorweiler - Updates made to handle redirects. Mind taking a look at the output of the parse.py script to see if the redirect is formatted correctly?
Thanks much!
Mike

**Comment by jdorweiler** [comment]
@mkorvink looks great!
https://beta.duckduckgo.com/?q=icd+a001&ia=about
https://beta.duckduckgo.com/?q=icd+a00.1&ia=about

Can you test it out and let me know what you think?  I added the trigger words `icd`, `icd10`, and `icd-10`

File: lib/DDG/Fathead/InternationalClassificationOfDiseases.PM [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package DDG::Fathead::InternationalClassificationOfDiseases;
```
Comment (jdorweiler) [inline lib/DDG/Fathead/InternationalClassificationOfDiseases.PM]: You can remove this file.  We're not using these anymore.

**Comment by mkorvink** [comment]
@jdorweiler - .PM File removed and beta site tested.  All looks good!

**Comment by jdorweiler** [comment]
:+1:  looks great!  I'll let you know when it's live.

**Comment by mkorvink** [comment]
Fantastic.  Thanks @jdorweiler and @moollaza for the help!

**Pull request merged**